<?xml version="1.0" ?>
<sdf version="1.10">
    <world name="moon_flat_world">

        <physics name="bullet_physics" type="bullet">
            <max_step_size>0.001</max_step_size>
            <real_time_factor>1.0</real_time_factor>

        </physics>
        <plugin 
            filename="gz-sim-physics-system"
            name="gz::sim::systems::Physics">
            <physics>bullet_physics</physics>
        </plugin>
        <plugin
            filename="gz-sim-user-commands-system"
            name="gz::sim::systems::UserCommands">
        </plugin>
        <plugin
            filename="gz-sim-scene-broadcaster-system"
            name="gz::sim::systems::SceneBroadcaster">
        </plugin>
        <plugin name="gz::sim::systems::Sensors" filename="gz-sim-sensors-system"/>
  <!-- plugin-specific params go here if needed -->

        
        <gravity>0 0 -1.62</gravity>

        <light type="directional" name="sun">
            <cast_shadows>true</cast_shadows>
            <pose>0 0 10 0 0 0</pose>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.2 0.2 0.2 1</specular>
            <attenuation>
                <range>1000</range>
                <constant>0.9</constant>
                <linear>0.01</linear>
                <quadratic>0.001</quadratic>
            </attenuation>
            <direction>-0.5 0.1 -0.9</direction>
        </light>

        <include>
          <uri>model://lander</uri>
          <pose>0 0 90 0 0 0</pose> <!-- Drop from 50m for IMU motion -->
        </include>
        
        <model name="ground_plane">
            <static>true</static>
            <link name="link">
            <collision name="collision">
            <geometry>
                <box>
                <size>100 100 0.1</size>  <!-- Large flat box as ground -->
                </box>
            </geometry>
            <surface>
                <bounce>
                <restitution_coefficient>0.5</restitution_coefficient>
                <threshold>1.0</threshold>
                </bounce>
                <friction>
                <ode>
                    <mu>0.1</mu>
                    <mu2>0.1</mu2>
                </ode>
                </friction>
            </surface>
            </collision>


                <visual name="visual">
                <geometry>
                    <plane>
                    <normal>0 0 1</normal>
                    <size>1000 1000</size>
                    </plane>
                </geometry>
                <material>
                    <ambient>0.8 0.8 0.8 1</ambient>
                    <diffuse>0.8 0.8 0.8 1</diffuse>
                    <specular>0.8 0.8 0.8 1</specular>
                </material>
                </visual>
            </link>
        </model>
    </world>
</sdf>